plugins/ca/root-ca/src/main/java/org/apache/cloudstack/ca/provider/RootCAProvider.java

```diff
@@ -365,8 +370,12 @@ private boolean loadManagementKeyStore() {
         if (managementKeyStore != null) {
             return true;
         }
-        final Certificate serverCertificate = issueCertificate(Collections.singletonList(NetUtils.getHostName()),
-                NetUtils.getAllDefaultNicIps(), getCaValidityDays());
+        List<String> nicIps = NetUtils.getAllDefaultNicIps();
```

Comment:
@JoaoJandre 
you can change the return type of the method `getAllDefaultNicIps` to `ArrayList<String>`

Reply:
I'm sorry, but I don't understand why I would change the return type from `List<String>` to `ArrayList<String>`. Could you explain?

Reply:
then it will not be needed

        nicIps = new ArrayList<>(new HashSet<>(nicIps));

Reply:
The idea behind this line is to make sure that all IPs in `nicIps` are unique. The `HashSet` constructor will enforce this, and then I convert it back to a list.

Reply:
Yes, in the `addConfiguredManagementIp` method (lines 402 through 409), I go through all of the interfaces and all configured IPs in those interfaces, and for each one, I check if they are in the configured `ManagementNetwork` range.